packages/iov-keycontrol/src/wallet.ts

```diff
@@ -94,3 +64,36 @@ export interface Wallet {
 
   readonly clone: () => Wallet;
```

Comment:
Clone a ReadonlyWallet gives you a mutable Wallet?
Is that safe?
Shall it return a ReadonlyWallet instead?

Reply:
Yes, because cloning the ReadonlyWallet does not change the original. So this is a way to make a mutable copy of the immutable object.

This property is required when adding a wallet to the UserProfile:

```
  public addWallet(wallet: Wallet): WalletInfo {
    if (!this.keyring) {
      throw new Error("UserProfile is currently locked");
    }

    const copy = wallet.clone();
    const info = this.keyring.add(copy);
    this.walletsProducer.update(this.walletInfos());
    return info;
  }
```

If the `copy` was immutable, the keyring could not perform the required changes on it.